qa/vagrant/src/main/java/org/elasticsearch/packaging/util/ServerUtils.java

```diff
@@ -0,0 +1,125 @@
+            try {
+
+                final HttpResponse response = Request.Get("http://localhost:9200/_cluster/health")
+                    .connectTimeout(120000)
+                    .socketTimeout(120000)
+                    .execute()
+                    .returnResponse();
+
+                if (response.getStatusLine().getStatusCode() >= 300) {
+                    final String statusLine = response.getStatusLine().toString();
+                    final String body = EntityUtils.toString(response.getEntity());
+                    throw new RuntimeException("Connecting to elasticsearch cluster health API failed:\n" + statusLine+ "\n" + body);
+                }
+
+                started = true;
+                break;
+
+            } catch (HttpHostConnectException e) {
+                // we want to retry if the connection is refused, so do nothing here
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+
+        }
+
+        if (started == false) {
+            throw new RuntimeException("Elasticsearch did not start");
+        }
+
+        final String url;
+        if (index == null) {
+            url = "http://localhost:9200/_cluster/health?wait_for_status=" + status + "&timeout=60s&pretty";
+        } else {
+            url = "http://localhost:9200/_cluster/health/" + index + "?wait_for_status=" + status + "&timeout=60s&pretty";
+
+        }
+
+        final String body = makeRequest(Request.Get(url));
+        assertThat("cluster health response must contain desired status", body, containsString(status));
+    }
+
+    public static void runElasticsearchTests() {
+        makeRequest(
+            Request.Post("http://localhost:9200/library/book/1?refresh=true&pretty")
+                .bodyString("{ \"title\": \"Book #1\", \"pages\": 123 }", ContentType.APPLICATION_JSON));
+
+        makeRequest(
+            Request.Post("http://localhost:9200/library/book/2?refresh=true&pretty")
+                .bodyString("{ \"title\": \"Book #2\", \"pages\": 456 }", ContentType.APPLICATION_JSON));
+
+        String count = makeRequest(Request.Get("http://localhost:9200/_count?pretty"));
+        assertThat(count, containsString("\"count\" : 2"));
+
+        makeRequest(Request.Delete("http://localhost:9200/_all"));
+    }
+
+    public static String makeRequest(Request request) {
+        try {
+            final HttpResponse response = request.execute().returnResponse();
+
+            final String body = EntityUtils.toString(response.getEntity());
+
+            if (response.getStatusLine().getStatusCode() >= 300) {
+                throw new RuntimeException("Request failed:\n" + response.getStatusLine().toString() + "\n" + body);
+            }
+
+            return body;
+        } catch (IOException e) {
```

Comment:
I tend to be fine declaring that my tests `throw IOException` or even `throw Exception`. So I would just let these bubble out.

Reply:
++